File: dev-docs/snippets/workflows/list-runs.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List runs for the current user, optionally filtered and paginated.
export async function listRuns(client: JsBaoClient, workflowKey: string) {
  // #region example
  const page = await client.workflows.listRuns({
    workflowKey,
    status: "completed",
    limit: 20,
  });
  // page.items: WorkflowRun[]  /  page.cursor for the next page
  // #endregion example
  return page;
}

File: dev-docs/snippets/workflows/list-runs.swift
import JsBaoClient

// List runs for the current user, optionally filtered and paginated.
func listRuns(client: JsBaoClient, workflowKey: String) async throws {
  // #region example
  let page = try await client.workflows.listRuns(
    options: ListWorkflowRunsOptions(
      workflowKey: workflowKey,
      status: "completed",
      limit: 20
    )
  )
  // page.items: [WorkflowRunInfo]  /  page.cursor for the next page
  // #endregion example
  _ = page
}
